amp-wistia-player extension

chen-anders opened the pull request:
Proposed `amp-wistia-player` component would embed a responsive Wistia player in an iframe. This single component would support publishers using Wistia.

Attributes specifying the media id will determine the embed URI that will be used. 

Example:

```
<amp-wistia-player 
    data-media-hashed-id="u8p9wq6mq8"
    width="512" 
    height="360">
</amp-wistia-player>
```

Above would yield http://fast.wistia.net/embed/iframe/u8p9wq6mq8

The component supports using play, pause, mute, unmute postMessage event to interact with the video.

MaxPower15 commented on examples/wistiaplayer.amp.html:
```diff
@@ -0,0 +1,18 @@
+    <amp-wistia-player
+      height=360
+      width=512
+      data-media-id="u8p9wq6mq8">
```
What do you think about changing this to `data-hashed-id`? I realize it's not super semantic, but it's the term we use in our docs. cc @mrdavidjcole for thought.

chen-anders replied:
Hm - we have multiple types of hashed_ids now (due to API v2) - how about `media-hashed-id`?

mrdavidjcole replied:
`media-hashed-id` makes sense to me.

MaxPower15 replied:
Not sure if this is obvious, but I think it still needs to start with `data-` to be valid markup--that's why all the others start with `data-` too. `data-video-hashed-id` is fine by me too.

chen-anders replied:
Yup - it would have to be `data-media-hashed-id`

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -0,0 +1,211 @@
+        this.element);
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
```
I would also add `scrolling="no"`, `allowtransparency=""`, and `title="Wistia video player"` based on https://github.com/wistia/wistia/blob/9e26649e785b816fcb39aa0d342ac3c9f995d992/lib/embed_codes/video/iframe_embed_code.rb#L5-L20.

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -0,0 +1,211 @@
+   */
+  sendCommand_(command) {
+    if (this.iframe_ && this.iframe_.contentWindow) {
+      this.iframe_.contentWindow./*OK*/postMessage(command, '*');
```
Any idea what that `/*OK*/` does?

chen-anders replied:
it's a comment; shouldn't do anything. I think it's just a note just to remind ourselves that it's authorized cross-site behaviour.

MaxPower15 replied:
Yeah, I guess I was just wondering if it's some compilation macro or something. Weird, haha.

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -0,0 +1,211 @@
+
+  /** @override */
+  showControls() {
+    // Not supported
```
We could support this pretty easily if we wanted, but I'm not sure why we'd want to.

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -0,0 +1,211 @@
+    if (!this.iframe_) {
+      return;
+    }
+    fullscreenEnter(dev().assertElement(this.iframe_));
```
Does this make the actual iframe fullscreen or act like a stub? Confused what's going on here. We have some logic that depends on knowing whether the video is in fullscreen mode or not, so we could notify the video when we're entering or exiting that mode, if that's what's happening.

chen-anders replied:
`fullScreenEnter` seems to call one of the browser native(?) methods: https://github.com/ampproject/amphtml/blob/master/src/dom.js#L793

Commit:
update README.md typo (#12640)

* update README.md typo

small typo in the email address

* Updating deadlink

Gregable committed:
Validator Rollup - pick up some test file changes (#12646)

* Add amp-list javascript extension to the amp-list test case.

* Add test file for missing amp-access extension.

jridgewell committed:
AMP Layers (#11830)

* Integrating with Resources

* Switch all rects to use relative to scroll positioning

* Complete demo

* Undo extension changes

These updates will be implemented per extension under the flag.

* Fix merge conflict

* Setup scrollable carousel to use layers conditionally

* Add layers experiment

* Remove unrelated changes

* Fixes

* Fix remeasuring children of layers that are already scrolled

* fixes for deeply disposing layers

* Remove residual layers experiment code

* Fix scrolled position calculations for root layers

* Speed up scrolling by lazy updating scroll position

* Fix layers in wrapped iOS

* Remove LayoutLayer

Instead, LayoutElements can handle its responsibilities with a boolean
flag, making our tree a little simpler to understand.

* Fixes, and remove static methods

* Ensure that root layers do not double count their scroll tops

Root layers (the browser's scroll layer in most cases) are the only elements that return both a `top` position based on the current scrollTop, and a non-negative scroll top (`$0.getBoundingClientRect().top === -$0.scrollTop`). All other elements don't double count this scroll top, since the scroll  position affects their children, not themselves.

This compensates.

* Ensure parent layer is refreshed after reparenting

* Add support for fixed position layers

* Make layers when we encounter a fixed position element

* Remove temporary hack

* Rename sizeWh and positionLt

* Have the viewport instantiate Layers

* Fix removing parent-layer-less elements

* Merge fixed position detection into loop

* Use Node#isConnected

* Fix getParentLayer

* Declare scrolling layer during creation

* Fix layers on non-iOS

* Ensure measurements are never mutated

* Use listen helper to handle passive

* Comments

* Comment and simplify

* Rename parent to ancestor

* Fix camel-casing in fields

* Reword comment

* Fix unintentional revert

* Test layers service

* Will's review

* Linting

* Review comments

* Lint

* Lint

* Move layers creation into Viewport service, not binding

* Fix test

Gregable committed:
Update validator release changelog. (#12647)

chen-anders committed:
gulpfile.js - Add amp-wistia-player

chen-anders committed:
Add AMP support for Wistia player

chen-anders committed:
Add amp-wistia-player example page

bradfrizzell committed:
Implement getCustomRealTimeConfigMacros for Doubleclick Fast Fetch (#12539)

* Start work on macros

* Add test coverage

* Responding to feedback

* Responded to feedback

* Respond to feedback

* Fix test

* Change url replacement to be cleaner

* Respond to feedback

* feedback

* Respond

chen-anders committed:
Replace 2017 with 2018 in Copyright Notice

chen-anders committed:
Fix lint errors

Gregable committed:
Render validator test case output inline. (#12648)

* Render validator test case output inline.

* Update nodejs files for inlined test output.

jridgewell committed:
Layers: Implement Disposable interface (#12653)

Fixes https://github.com/ampproject/amphtml/issues/12553

jridgewell committed:
Prevent unbounded accumulated error messages (#12585)

Hopefully fixes the `PayloadTooLargeError: request entity too large`
messages we've been seeing.

chen-anders committed:
Listen to and handle events from player

zhouyx committed:
fix (#12657)

alanorozco committed:
Refactor pagination buttons and change based on navigation state (#12511)

alanorozco committed:
Polish and optimize desktop background (#12629)

jonkeller committed:
Fix commas (#12660)

rsimha committed:
Move react-dates from devDependencies to dependencies (#12571)

sklobovskaya committed:
amp-gwd-animation: Fix timeline-event-prefix attribute validator rule (#12664)

rsimha committed:
Change default user logging level of AMP runtime from `OFF` to `WARN` (#12649)

tbaik committed:
Add class bindings support for SVG tags (#12643)

rsimha committed:
Write prod config and enable local dev for gulp watch (#12665)

prateekbh committed:
rotate bug fix (#12642)

* rotate bug fix

* fixing comments

prateekbh committed:
no forced repaint on ipad (#12658)

prateekbh committed:
Speaker fix (#12611)

* minor animation fixes

* bug fix for education on desktop

* speaker icon beside progress bar

* making system layer adopt flexbox on desktop

* clear element location

cathyxz committed:
Update amp-image-lightbox default object-fit and zoom upper bound for extreme aspect ratios (#12636)

* Update width and height for amp-image-lightbox to accomodate extreme ratios

* Add manual tests for amp-image-lightbox extreme aspect ratios

* Adjust max scale based on aspect ratio

* Address incorrect use of typecheck

rsimha committed:
Simplify isLocalDev logic in mode.js (#12667)

cvializ committed:
Fix name mangling bug. Update react-dates build. (#12672)

glevitzky committed:
Updates Refresh's network-level docs (#12013)

noranazmy committed:
Don't load responsive ad if tag height is not AdSense whitelisted height (#12225)

prateekbh committed:
Change education hint for no previous page and remove text from education hint (#12623)

* minor animation fixes

* bug fix for education on desktop

* changing no previous page hints

* speeed to 275

* fixing arrow icons size

* adding text to hint overlay

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -76,9 +93,23 @@ class AmpWistiaPlayer extends AMP.BaseElement {
+      this.listenToFrame_();
+      this.element.dispatchCustomEvent(VideoEvents.LOAD);
+    });
+    this.playerReadyResolver_(loaded);
```
It looks like this is being resolved when the iframe loads instead of when the player reports "ready." Am I reading it correctly? That might be fine functionally, but if I did read it right, I would rename it to avoid confusion. Maybe `playerLoadedPromise_` and `playerLoadedResolver_`?

chen-anders replied:
I've been trying to keep things consistent with https://github.com/ampproject/amphtml/blob/master/extensions/amp-youtube/0.1/amp-youtube.js

MaxPower15 replied:
Gotcha. I think this is somewhere we should depart from their language though--we have no obligation to have the same API as youtube. I think, for our maintenance, it makes the most sense to stay consistent with the rest of our code base.

chen-anders replied:
Will do

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -88,25 +119,88 @@ class AmpWistiaPlayer extends AMP.BaseElement {
 
+  /** @private */
+  handleWistiaMessages_(event) {
+    if (event.origin !== 'https://fast.wistia.net' ||
```
Do we need to do the origin check? Would be hard to test this with local embeds, and sometimes they're hosted on fast.wistia.com or fast.wistia.st too.

chen-anders replied:
My take is that this is easily changeable when amphtml is in local dev since we won't be able to technically insert a different origin from what is on google's amp servers (prod).

MaxPower15 replied:
Point taken considering the maintenance perspective. 👍

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -76,9 +93,23 @@ class AmpWistiaPlayer extends AMP.BaseElement {
     iframe.src = '//fast.wistia.net/embed/iframe/' + encodeURIComponent(
```
Would be cool if this was a configurable option in the amp tag, for local and staging testing, now that I'm thinking about it.

chen-anders replied:
I'm very sure that google maintainers ain't going to like that.

MaxPower15 replied:
Mm, good point.

MaxPower15 commented on extensions/amp-wistia-player/0.1/amp-wistia-player.js:
```diff
@@ -88,25 +119,88 @@ class AmpWistiaPlayer extends AMP.BaseElement {
+        this.element.dispatchCustomEvent(VideoEvents.PAUSE);
+        this.element.dispatchCustomEvent(VideoEvents.ENDED);
+      }
+    } else if (data['method'] == 'mutechange') {
```
This is kinda semantics, but to me, `mutechange` is a player event. I think it confuses me because our iframe API passes an argument called `method` which actually causes a method to be called.

chen-anders replied:
okay I suppose I could make a second postMessage thing inside `_inside_iframe.coffee` that passes messages in a slightly different format rather than re-using the pre-existing one.

MaxPower15 replied:
Which pre-existing format are you referring to--the one in amp or the one in our player? Just a bit confused. Any relevant code links?

chen-anders replied:
https://github.com/wistia/wistia/blob/anders/add-amp-support-to-iframe-embed/player/modern/src/external/E-v1/_inside_iframe.coffee#L102

MaxPower15 replied:
Ah, I see what you're looking at now. That's that `method` code I was talking about too. It would literally call the `updateProperties` method on the iframe API object. `_isMuted` gets updated like so: https://github.com/wistia/wistia/blob/c8a7e1ce3ceaad43d6854dd4f2691c3c60230ac5/player/modern/src/external/E-v1/_inside_iframe.coffee#L130-L131

If you want to leverage the existing events, you can look at _trigger, which is what we use to post up most events from the player: https://github.com/wistia/wistia/blob/c8a7e1ce3ceaad43d6854dd4f2691c3c60230ac5/player/modern/src/external/E-v1/_inside_iframe.coffee#L169.

MaxPower15 replied:
To follow up, I think that, as it stands, this code would cause an error if E-v1 was on the same page as the AMP script because the `mutechange` method is not defined on the iframe api object: https://github.com/wistia/wistia/blob/c8a7e1ce3ceaad43d6854dd4f2691c3c60230ac5/player/modern/src/external/E-v1/_iframe_api.coffee#L659

cvializ committed:
Use simple a11y class and offset. (#12488)

* Use simple a11y class and offset.

* Update test, fix bug

rsimha committed:
Fix ads served from non-localhost servers (#12673)

rsimha committed:
Eliminate the use of String.padStart() (#12681)

rsimha committed:
Eliminate the use of Object.values() (#12686)

jonkeller committed:
Changes warning to error (#12678)

* Changes warning to error

* Updates unit test

monish-chhadwa committed:
Register medianet as a vendor for doubleclick RTC (#12676)

jamesshannon committed:
Copy title attribute from component tag to iframe (#12617)

zhouyx committed:
use dev().error to report (#12684)

calebcordry committed:
allow copyright 2018 (#12700)

mrjoro committed:
Changes to the new status label (#12492)

mrjoro committed:
change to new "Status Update" label (#12491)

Commit:
Don't rewrite values of binding attributes in sanitizer.js (#12688)

* fix incorrect rewriting of [href] attr vals in sanitizer

* lint

Commit:
Fix several amp-list/amp-bind interaction bugs (#12687)

* fix several amp-list/amp-bind composed bugs

* lint

* add amp-list unit test and a lot of clean up

* replace bound fns with closures

* lint

* daniel's comments

* simplify pre-layout unit test

noranazmy committed:
Don't load full-width responsive ad if tag width is not 100vw (#12695)

donttrustthisbot committed:
cron job from @erwinmombay to update size.txt and size.csv (#12606)

rsimha committed:
Write AMP_CONFIG to 3p frame to enable local development (#12703)

donttrustthisbot committed:
cron job from @erwinmombay to update size.txt and size.csv (#12709)

Gregable committed:
Validator rollup (#12704)

* Validator Refactor: Add the best matching ParsedTagSpecs to the tag stack. This paves the way for us the ParsedTagSpec data for matching children.

* Revision bump for #12664

* Disallow template tags on children of `<div submit-success template=...>` tags.

* Revision bump for #12617

rsimha committed:
Allow pr-check.js to be run locally (#12711)

rsimha committed:
Exclude generated dirs in validator/webui from presubmit checks (#12714)

glevitzky committed:
Update refresh doc to reflect full launch. (#12718)

rsimha committed:
Initial set up for greenkeeper-lockfile (#12683)

greenkeeper[bot] committed:
Update dependencies to enable Greenkeeper 🌴 (#12679)

noranazmy committed:
Remove breaking width check in 3p path (#12722)

rsimha committed:
Update yarn.lock (#12720)

jonkeller committed:
Long task warn to err (#12705)

* Changes warning to error

* Updates unit test

* Revert "Updates unit test"

This reverts commit 7cc210742c2abfcbcd76fbc4a46ad8829b24a0a3.

* Log less info upon long task violation

* Un-revert unit test changes

* Un-removes 3 metrics because Malte said they're ok

alanorozco committed:
Prevent locking orientation from blocking rendering on error (#12662)

alanorozco committed:
Remove color section from WhatsApp icon (#12627)

rsimha committed:
Add a wrapper 'gulp pr-check' task for pr-check.js (#12715)

rsimha committed:
Update GH_TOKEN for greenkeeper (#12732)

alanorozco committed:
Set of amp-story bookend fixes (#12661)

* Use CSS transition for toggle. Animating by JS is no longer needed.
* Fixes replay. Broken when made async on #12511
* Fixes page blur bleed. Broken by #12632

honeybadgerdontcare committed:
Validator Chrome Extension: handle AMP Validator JavaScript not loading. (#12733)

* Handle not loading AMP Validator gracefully in chrome extension.

* Handle not loading AMP Validator gracefully in chrome extension.

* clean up

* clean up

bradfrizzell committed:
Add TGT macro for targeting data in json (#12707)

* Add TGT macro for targeting data in json

* Respond to feedback

* golf

* Fix

* fix check-types

* fix

rsimha committed:
Remove phantomjs-prebuilt from dev deps (#12737)

zhouyx committed:
remove experimental flags (#12735)

lannka committed:
Add canonicalOrigin support to the CID API. (#12716)

powdercloud committed:
Apply the contents of the "ads." file as update to ads.html, as intended. (#12739)

* Apply the contents of the "ads." file as update to ads.html, as intended.

* fix output

monish-chhadwa committed:
Extract publisher domain from canonical url (#12694)

cvializ committed:
Add tests for dates-list.js (#12701)

* Add tests for dates-list.js

* Update react-utils.js

Fix if->boolean mistake

zhouyx committed:
check for parentElement for sandbox analytics (#12746)

rsimha committed:
Add a version check + warning for node (#12745)

alanorozco committed:
Factor-out scrolling logic and styling for share widget (#12628)

cathyxz committed:
Return sidebar focus to opener after close v2 (#12601)

* Update manual test for amp sidebar to show breaking scroll

* Add integration tests for sidebar focus

* Return sidebar focus to opener after close

* Remove unnecessary fixture and unit test for sidebar focus

* Fix build

* Fix tests after aria-hidden resequence

* Address comments

* Undo reorder of aria-hidden

* Fix sidebar scroll test

* Remove unnecessary clock tick

* Move amp-sidebar integration test to proper folder

* Correct path for event helper post file move

* Remove unnecessary boilerplate code

* Update copyright year to 2018

* Revert test back to using poll, remove dispatchCustomEvent

* Remove control for scroll position

* Extend timeout for flakes on windows environments

cathyxz committed:
Get lightbox to work with group ids (#12663)

* Implement mapping for lightbox groups without killing existing functionality

* Draft work for getting lightbox groups to actually work

* Skip auto lightbox tests

* Reuse instead of rebuild gallery

* Reuse thumbnails instead of rebuilding them

* Address comments

* Change to use attribute amp-lightbox-group instead of directly using an id

* Add types and scopedQuerySelector

* Fix dev assert

Commit:
Add animations to amp-story spec (#12744)

zhouyx committed:
visibity-model-bug-fix (#12752)

zhouyx committed:
Use try catch to protect querySelector (#12734)

* use try catch to protect querySelector

* only send one error

* reorder

OmriKeret committed:
<amp-apester-media> fix random embed and improve performance (#11936)

rsimha committed:
Fix LGTM alerts in validator.js (#12758)

honeybadgerdontcare committed:
Fix lgtm errors in htmlparser.js (#12764)

aghassemi committed:
amp-lightbox: Better resource scheduling for scrollable lightbox (#12751)

rsimha committed:
Fix LGTM errors in src/service/ (#12767)

rsimha committed:
Fix LGTM errors in build-system/ (#12774)

rsimha committed:
Fix LGTM errors in extensions/ (#12770)

zhouyx committed:
remove unused code (#12776)

rsimha committed:
Fix LGTM error in validator/chromeextension/popup-validator.html (#12781)

zhouyx committed:
clean up tracker for document not visible error (#12698)

rsimha committed:
Fix LGTM error in gulpfile.js (#12778)

honeybadgerdontcare committed:
Validator Chrome Extension: Use tabs UserAgent in requests (#12750)

* Validator Chrome Extension: Use tabs UserAgent in requests

* clean up

rsimha committed:
Fix LGTM errors in error.js (#12782)

rsimha committed:
Fix LGTM errors in 3p code (#12785)

rsimha committed:
Fix LGTM errors in examples/ (#12777)

rsimha committed:
Fix greenkeeper-lockfile on Travis (#12789)

rsimha committed:
Set yarn path before installing greenkeeper-lockfile (#12791)

cvializ committed:
Swap preact back in with race condition band-aided (#12784)

* Swap preact back in with race condition band-aided

* Remove React and react-dom

alanorozco committed:
Roll forward MediaPool changes and fix issues (#12604)

mrjoro committed:
make the corp CLA instructions more clear (#12795)

gmajoulet committed:
Preloading story bookend (#12783)

* Maybe preloading the bookend.

* Bookend tests + re enabling some old tests.

* Pre rendering the bookend.

* Lint.

newmuis committed:
Remove fullscreen from amp-story (#12772)

* Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events

* Revert "Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events"

This reverts commit 0500899056e84a85138352aa6ea7edc8d9e033d0.

* Remove fullscreen

* Remove experiment

* Remove fullscreen

* Remove experiment

* Remove unused functions

cathyxz committed:
Implement lightbox attribute for carousels (#12771)

* Draft work on getting carousels to work with the lightbox attribute

* Address comments

* Use signals for detecting carousel load in amp-lightbox-viewer as well

* Change scopedQuerySelectorAll to querySelectorAll for perf benefit

alanorozco committed:
Trigger slidechange event from dynamic calls (#12814)

uniquelygeneric committed:
Added Nielsen support for EU (#12702)

* Added support for EU

* Added support for EU

* Added support for the EU

* Added support for EU

alanorozco committed:
Load amp-social-share when system share is supported. (#12812)

cvializ committed:
Add linter rule to prevent and fix redundant non-nullable primtives (#12809)

* Add linter error for primitives specified as non-nullable

* Fix redundant non-nullable primitives

* Add symbol to list of non-nullable primitives. Rename

* Add function and undefined non-nullability checks

* Fix new linter errors

rsimha committed:
Set up Greenkeeper GH_TOKEN for PR builds (#12807)

jonkeller committed:
Only preload iframe transport client lib if needed (#12779)

* Delays preload of iframe transport client lib until we know iframe transport is being used

* Fixes unit test

* Removes debugger stmt

* Fixes linter errors

* Stubs getting resource ID to avoid catch

* Stubs getting resource ID to avoid catch

* No-op commit to force re-run of CLA checker

* Spies on preload, addresses other nits

* Fixes method signature

* Combines 2 expect()s into one

bradfrizzell committed:
Allow objects to be given as macro values in RTC config (#12802)

* Allow objects to be given as macro values, just stringify 'em

* Update docs

* Don't loop over the same thing twice

prateekbh committed:
no amp story on vertical swipe (#12749)

prateekbh committed:
Hide text/icon on no previous page hint  (#12796)

* no text or icon on no back hint

* Update amp-story-hint.css

cvializ committed:
Fix primitive wrappers in Closure type tags (#12820)

* Forbid primitive wrapper types in CC type tags

* Fix primitive wrappers in CC type tags

danielrozenberg committed:
Update third_party/caja to v6012 (#12680)

* Updated compile.sh to maintain required modifications
* Removed outdated modifications from the list in README.md

prateekbh committed:
No previous from tap on card on desktop (#12671)

* no back from card tap on desktop

* fixing types

* refactor manual advancement

* changing names

* explicit previous check

rsimha committed:
Remove greenkeeper token  (#12822)

* Replace old greenkeeper token with a new encrypted token

* Remove JWT encrypted token

Commit:
Yieldpro support (#12773)

* add ancora ads

* remove publicate (test)value

* rebrand ancora to yieldpro

* rebrand ancora to yieldpro for amp-ad.md as well

* remove ancora js & md

* resubmit after signed CLA

clawr committed:
amp-ad-exit layout validation (#12600)

* Allow "nodisplay" and unspecified ("container") for amp-ad-exit's layout. See #12022.

Tested:
gulp validator

* document that layout is optional

rsimha committed:
Fix all remaining LGTM warnings in AMP code (#12826)

donttrustthisbot committed:
cron job from @erwinmombay to update size.txt and size.csv (#12833)

adventive-engineering committed:
Adventive amp-ad type (#12742)

* Alphabetized lists

* Adding Adventive ad type implementation.

Modified files listed in `ads/README.md`

honeybadgerdontcare committed:
Remove unnecessary pre-compiled file from package. (#12790)

shawnbuso committed:
amp-ima-video: Ignore tap-and-drag to start video. Only start on tap without drag (#12803)

aghassemi committed:
amp-list: remove `src` when local data is used to render (#12861)

honeybadgerdontcare committed:
Update amp4ads-amp-ad-exit test output. (#12846)

bradfrizzell committed:
Change how AMP inabox validates that the source of a postMessage is valid (#12541)

* Allow nested frames if they are friendly

* Add tests

* Respond to feedback

* Respond to feedback

* Respond to feedback

* Fix dumb dumb

* small fix

* See...my....test! See my test! Made of real gorilla chest

* Continued work

* More work in progress

* Continued work, added getMeasureableFrame

* WIP - fixing tests

* Change how getFrameElement works

* Fix check-types and tests

* Respond to feedback

* Make getMeasureableFrame clearer

* minor changes

* Respond to feedback

* Respond to feedback

prateekbh committed:
add on tap listener for multiple advance (#12844)

prateekbh committed:
No padding, no back gradient (#12845)

zhouyx committed:
Remove ampcors check when fetching analytics remote config (#12787)

danielrozenberg committed:
Rewrite third_party/caja/compile.sh to use patches where possible (#12851)

bradfrizzell committed:
Fix macros for RTC Doubleclick (#12841)

* fix macros

* respond to feedback

* Stringify

* Fix tests

zhouyx committed:
unskip test that fail with lolex v2 (#12858)

cathyxz committed:
Pull amp-image-viewer out into it's own component (#12810)

* Make image viewer its own component

* Hack prototype of lightbox-viewer that uses the amp-image-viewer component

* Remove unnecessary files

* Add scheduleLayout changes

* Remove manual test

* Remove image-viewer library

* Update todos

* Fix code quality

* Add pause and resume callbacks

* Add copyright and fix bad variable naming

* Fix hasOwnProperty --> hasAttribute

rsimha committed:
Remove the soon-to-be-deprecated gulp-util (#12825)

zhouyx committed:
remove old error logging (#12875)

cathyxz committed:
Use the proper getAttribute function (#12872)

newmuis committed:
Show the audio icon if there are media elements or background audio in the story (#12763)

* Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events

* Revert "Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events"

This reverts commit 0500899056e84a85138352aa6ea7edc8d9e033d0.

* Show the audio icon if and only if there is media present in the story.

* Add useless whitespace to re-trigger checks

* Remove useless whitespace

* Remove obsolete function

rsimha committed:
Start gulp server after build is complete (#12882)

prateekbh committed:
fix animating circle fixed position (#12871)

nainar committed:
[amp-audio] Adds preload attribute (#12832)

danielrozenberg committed:
Referenced templates also affect [submit-error] (#12813)

* Referenced templates also affect [submit-error]

* Address @honeybadgerdontcare's comment

* Remove dispatch_key from both submit-X specs

newmuis committed:
Bless amp-story media on user navigation (#12892)

* Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events

* Revert "Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events"

This reverts commit 0500899056e84a85138352aa6ea7edc8d9e033d0.

* Bless media on user gesture navigation

* Update comments for bless_ and blessAll functions

cvializ committed:
Fix carousel arrow disappearing on desktop without controls attr (#12894)

* Fix carousel arrow disappearing on desktop without controls attr

* Install input service for tests

jridgewell committed:
Add Sidebar to FixedLayer after displaying (#12899)

* Add Sidebar to FixedLayer after displaying

Fixes #12891.

* Ensure display:none elements aren't manually added to fixed-layer

cathyxz committed:
Add lightbox-thumbnail-src attribute (#12877)

cathyxz committed:
Auto install lightbox without adding a <amp-lightbox-viewer> tag (#12873)

* Auto install lightbox without tag

* Swap waitForBodyPromise for whenBodyAvailable

cathyxz committed:
[Prototype] Add exit animations to lightbox viewer back to source element (#12879)

* Add exit animations to lightbox viewer back to source element

* Refactor transition duration calculation into own function

Gregable committed:
Support 'geo:' URL scheme for anchor href attributes. (#12904)

Reference: https://en.wikipedia.org/wiki/Geo_URI_scheme

Fixes: https://github.com/ampproject/amphtml/issues/12888

lannka committed:
Print CID API error response to browser console. (#12738)

* Print CID API error response to browser console.

* Fix test

rsimha committed:
Consolidate gulp watch and gulp build (#12900)

misscs committed:
Add instructions for using Typekit (#12602)

zhucl committed:
Update the a4a spec to require amp-animation based on (#11769)

aghassemi committed:
amp-carousel : install input service before using it (#12910)

chen-anders committed:
Prefix state/isMuted with const in handleWistiaMessages

newmuis committed:
amp-story initial load tweaks (#12842)

* Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events

* Revert "Change progress bar to not handle active page, since active page updates progress bar through PAGE_PROGRESS events"

This reverts commit 0500899056e84a85138352aa6ea7edc8d9e033d0.

* Modify initial loading styles and signals.

* Fix lint and types

* Change loading logic to depend on canplay events for media.  Suppress errors to ensure all elements have a chance to load.

* Refactor page load utility functions in amp-story

* Fix lint

* Move dispatch out of vsync

* Add whitespace to trigger Travis re-run

* Remove unnecessary whitespace

* Rename loadPromise vars

* Fix line length

rsimha committed:
Add missing option to buildExtensions() in gulp watch / gulp build (#12911)

sanjsanj committed:
amp-story progress-bar segment fills for current page (#12613)

* Pages without auto-advance-after attr fill their progress bar immediately

* can progress forward properly for manual, time-based and media-based elements

* correctly finds amp-video media elements

* be more explicit about amp-video scopedQuerySelector

* rename media.html to progress-bar.html and use it to test aspects of progress-bar

* create a media-element constant

* replaying story does not trigger active event on last page before triggering again on first page

* remove comment

* remove comment

* improve jsdoc

* improve jsdoc description of manual advancement class

* reinstantiate multiple configs so that navigational clicks can occur on devices

* can navigate all page types on emulated devices

* does not update manual progress segment if other progression modes are available

* fix eslint error

* merge master, remove unneeded check for onTapListeners

* remove unneeded import

* remove unneeded comment

* progress update moved to base class start method

* revert change

* remove MEDIA_ELEMENT_SELECTOR const and code that used it

cathyxz committed:
Add media queries with css grid layout to ensure responsive gallery (#12905)

* Add media queries with css grid layout to ensure responsive gallery

* Tweak media queries and repeat bounds

* Remove dependency on vw and use padding-top instead

cvializ committed:
Fix double-trigger video-ended in analytics (#12922)

rsimha committed:
Fix error reporting in compileJs (#12928)

calebcordry committed:
Support nested URL macros (#12682)

* parser first draft

* WIP merges

* parser finally working 👍

* get rid of sync support

* refactor to class

* writing tests

* move files and fix bugs

* add experiment flag

* clean up diff

* fix experiment path

* pass correct win & lint

* more linting

* support double commas

* bug fix - all tests passing

* code review

* lint

* migrate tests

* code review

* linter

* code review

* remove unecessary sorting

* type checks

danielrozenberg committed:
Allow <form> and nested <template>s in <template>s (#12853)

mrjoro committed:
document our new design review time rotation (#12942)

chen-anders committed:
Fallback to "Wistia Video Player" title attribute if it does not exist

chen-anders committed:
Use https for iframe.src

chen-anders committed:
Add layout=fixed into Wistia Player example

chen-anders committed:
Make message handling consistent with non-AMP iframes

chen-anders committed:
Fix lint errors

chen-anders committed:
Add amp-wistia-player to whitelist in dep-check-config

chen-anders committed:
Update amp-wistia-player test for iframe.src being HTTPS only
